ibis/backends/duckdb/__init__.py

```diff
@@ -1145,57 +1148,154 @@ def to_xlsx(
         self.con.execute(copy_cmd).fetchall()
 
     def attach(
-        self, path: str | Path, name: str | None = None, read_only: bool = False
+        self,
+        path_or_url: str | Path,
+        /,
+        *,
+        name: str | None = None,
+        on_exists: Literal["error", "ignore", "replace"] = "error",
+        read_only: bool = False,
+        type: Literal["duckdb", "sqlite", "postgres", "mysql"] | None = None,
+        more_options: Iterable[str] = [],
```

Comment:
Why not use a mapping here? That seems like the most natural fit for a bag of options?

What purpose is `more` serving here in the API? It's at best redundant IMO (we know that the next argument is "more" arguments). Let's call this just `options`.

And this definitely cannot default to anything mutable like a `list`. The empty `tuple` is fine though.

Reply:
I moved to `options: Mapping[str, Any] = {}`. It actually is fine for it to be mutable as long as we don't modify it within the method, and I think {} as a default is more clear than something immutable like None.